tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -246,3 +247,144 @@ func createPvcInParallel(client clientset.Interface, namespace string, diskSize
+	}
+
+	for _, x := range siteHosts {
+		siteEsxMap[x] = true
+	}
+	allMasterIps := getK8sMasterIPs(ctx, client)
+	framework.Logf("all master ips : %v", allMasterIps)
+	sshClientConfig := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.Password(k8sVmPasswd),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	framework.Logf("Site esx map : %v", siteEsxMap)
+	// Assuming atleast one master is on that site
+	vcAddress := e2eVSphere.Config.Global.VCenterHostname
+	for _, masterIp := range allMasterIps {
+		cmd := "export GOVC_INSECURE=1;"
+		cmd += fmt.Sprintf("export GOVC_URL='https://administrator@vsphere.local:Admin!23@%s';", vcAddress)
+		cmd += fmt.Sprintf("govc vm.info --vm.ip=%s;", masterIp)
+		result, err := sshExec(sshClientConfig, allMasterIps[0], cmd)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		hostIp := strings.Split(result.Stdout, "Host:")
+		host := strings.TrimSpace(hostIp[1])
+		if siteEsxMap[host] {
+			masterIpOnSite = masterIp
+			break
+		}
+	}
+	framework.Logf("Master IP on site : %s", masterIpOnSite)
+	if masterIpOnSite != "" {
+		return masterIpOnSite, nil
+	} else {
+		return "", fmt.Errorf("couldn't find a master running on site")
+	}
+}
+
+// changeLeaderOfContainerToComeUpOnMaster ensures that the leader of a container comes up on
+// a specific master node on that site
+func changeLeaderOfContainerToComeUpOnMaster(ctx context.Context, client clientset.Interface,
+	sshClientConfig *ssh.ClientConfig, csiContainerName string) error {
+	// Fetch the Master Ip on that site
+	masterIpOnSite, err := getMasterIpOnSite(ctx, client, false)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	// Get the master Ip where leader of csi container is running
+	allMasterIps := getK8sMasterIPs(ctx, client)
+	// Remove master ip which is on that site from list of master ip of all nodes
+	for i, v := range allMasterIps {
+		if v == masterIpOnSite {
+			allMasterIps = append(allMasterIps[:i], allMasterIps[i+1:]...)
+			break
+		}
+	}
+
+	leaderFoundOnsite := false
+	waitErr := wait.PollImmediate(healthStatusPollInterval, pollTimeout, func() (bool, error) {
+		// Check if leader of csi container comes up on master node of secondary site
+		_, masterIp, err := getK8sMasterNodeIPWhereControllerLeaderIsRunning(ctx, client, sshClientConfig,
+			csiContainerName)
+		framework.Logf("Leader is %s", masterIp)
+		if err != nil {
+			return false, err
+		}
+		// Pause and kill container of csi container on other master nodes
+		if masterIp == masterIpOnSite {
+			leaderFoundOnsite = true
+			framework.Logf("Leader of %s found on site", csiContainerName)
+			return true, nil
+		}
+		err = executeDockerPauseKillCmd(sshClientConfig, allMasterIps[0], csiContainerName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = executeDockerPauseKillCmd(sshClientConfig, allMasterIps[1], csiContainerName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		return false, nil
+	})
+	if leaderFoundOnsite {
```

Comment:
This will be flagged by golint, please run `make check`
you can change it to,

```golang
if !leaderFoundOnsite {return fmt.Errorf(...) }
return waitErr
```